### PR TITLE
Added /api/registration endpoint

### PR DESCRIPTION
**Pull request by Nomi**

### CODE REVIEW

File: server/endpoints/registration.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        const data = this.parseAndValidate(req.body, registrationValidation, true);
+        checkUsernameUniqueness(req,data.username);
+        var usrIns= new User(null,true);
+        usrIns.username=data.username;
+        usrIns.password=await hashPassword(data.password);
+        usrIns.email=data.email;
+        usrIns.created=Date.now();
+        usrIns.isAdmin="0";
+        usrIns.insert(req.database);
+        return usrIns.serialize();
```
Comment (MateuszWojtynski) [inline server/endpoints/registration.js:29]: Before you return user object, remove password from `usrIns`.

`usrIns = usrIns.serialize();`
` delete usrIns.password;`
`return usrIns;`

File: server/endpoints/registration.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        this.enforceContentType(req, 'application/json');
+        const data = this.parseAndValidate(req.body, registrationValidation, true);
+        checkUsernameUniqueness(req,data.username);
+        var usrIns= new User(null,true);
+        usrIns.username=data.username;
+        usrIns.password=await hashPassword(data.password);
+        usrIns.email=data.email;
+        usrIns.created=Date.now();
+        usrIns.isAdmin="0";
+        usrIns.insert(req.database);
```
Comment (MateuszWojtynski) [inline server/endpoints/registration.js:28]: Check if insertion succeeded.

`if(!usrIns.id)`
`{`
 `throw new Error('User creation failed,');`
`}`

**Review by MateuszWojtynski — CHANGES_REQUESTED** [review]

File: server/endpoints/registration.js (hunk 1 above)
Comment (Nomi) [reply server/endpoints/registration.js:29]: Implemented. BTW, the order of serialization and deletion might be wrong in the given snippets.

File: server/endpoints/registration.js (hunk 2 above)
Comment (Nomi) [reply server/endpoints/registration.js:28]: done

File: server/endpoints/registration.js (hunk 1 above)
Comment (MateuszWojtynski) [reply server/endpoints/registration.js]: True

**Review by MateuszWojtynski — APPROVED** [review]

**Pull request merged**